Pcap++/header/PcapFilter.h

```diff
@@ -3,10 +3,14 @@
 
 #include <string>
 #include <vector>
+#include <memory>
 #include "ProtocolType.h"
 #include <stdint.h>
 #include "ArpLayer.h"
 
+//Required for FilterTester
+struct bpf_program;

```

Comment:
can we hide this structure and move it to the implementation file?

instead you can have the `prog` private member be of type `void *` and do the casting in the implementation file

Reply:
I can do this, I am just accustomed to keeping type safety, but in this case its all internal so it doesn't TECHNICALLY void the safety as external users can't touch this.